content/2022/hosts.md

```diff
@@ -0,0 +1,14 @@
+---
+title: "Hosts"
+seo_title: "Hosts | JakartaOne Livestream 2022" 
+date: 2022-09-16T09:31:33-04:00

```

Comment:
The date is in the future, just making sure it's intended.

Reply:
I'm not too sure on how we usually handle the hugo's date front-matter...

I know the most common use case of `date` with Hugo sites in general is to control when to display content. Like, you would make the `date` of a blog post in the future to prevent it from being generated before that date.

I've simply just used them as a "created-at" timestamp of the file itself. I noticed a lot of other pages do the same.

Reply:
You are right, putting a date in the future will prevent the page from being displayed on the site.

But I think that if we merge this right now, the new page will not just show up tomorrow morning as Hugo would need to rebuild the static website to include the new page.

Reply:
Oh I see what you're saying... but this is the 16th of Sept, not Nov.

Reply:
# Difference Between Various Front Matter Date Variables

## date

the datetime assigned to this page. This is usually fetched from the date field in front matter, but this behavior is configurable.

## publishDate

if in the future, content will not be rendered unless the --buildFuture flag is passed to hugo.

## lastmod

the datetime at which the content was last modified.

## expiryDate

the datetime at which the content should no longer be published by Hugo; expired content will not be rendered unless the --buildExpired flag is passed to the hugo command.

See https://gohugo.io/content-management/front-matter/


---

For events, the value of `Date` should be the start date of the event. We can leverage the `publishDate` to control when to publish a page.

We use the date to order events on list pages such as https://events.eclipse.org/2022/

More on ordering: https://gohugo.io/templates/lists/#order-content

If we need to do that, we can configure the CI platform (netlify/jenkins) to rebuild the site once a day to ensure that a page appears on the specified publishing data.

Reply:
@oliviergoulet5 oh wow, I miss read that one.. I thought it was november.

Thanks @chrisguindon for the clarification.

Reply:
Updated the date to reflect the date of the event.